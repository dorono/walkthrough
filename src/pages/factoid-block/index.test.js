import React from 'react';
import {shallow} from 'enzyme';
import {FactoidBlockPage} from './index';

const defaultProps = {
    data: {
        transactions: [],
    },
    location: {},
};

describe('FactoidBlock', () => {
    it('should render without errors', () => {
        shallow(<FactoidBlockPage {...defaultProps} />);
    });

    it('should match snapshot', () => {
        expect(shallow(<FactoidBlockPage {...defaultProps} />)).toMatchSnapshot();
    });
});
