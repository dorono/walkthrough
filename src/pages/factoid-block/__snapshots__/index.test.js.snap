// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FactoidBlock should match snapshot 1`] = `
<div>
  <Container
    primary={true}
    showFullWidthBanner={false}
    title="Factoid block"
  >
    <VerticalToHorizontal
      verticalUpTo="small"
    >
      <Vertical>
        <Box
          type="outline"
        >
          <Vertical>
            <div>
              <Label>
                INPUTS
              </Label>
              <Amount
                unit="FCT"
              />
            </div>
            <div>
              <Label>
                OUTPUTS
              </Label>
              <Amount
                unit="FCT"
              />
            </div>
            <div>
              <Label>
                ECS CREATED
              </Label>
              <Amount
                unit="EC"
              />
            </div>
          </Vertical>
        </Box>
      </Vertical>
      <Vertical>
        <Box
          type="outline"
        >
          <Label>
            KEYMR
          </Label>
          <withRouter(Hash)
            type="fblock"
          />
        </Box>
        <Box
          type="fill"
        >
          <Label>
            PARENT DIRECTORY BLOCK
          </Label>
          <DirectoryBlockLink />
        </Box>
      </Vertical>
    </VerticalToHorizontal>
  </Container>
  <Sortable
    items={Array []}
    sortOptions={
      Array [
        undefined,
        undefined,
        Object {
          "func": [Function],
          "label": "Highest input first",
        },
        Object {
          "func": [Function],
          "label": "Lowest input first",
        },
        Object {
          "func": [Function],
          "label": "Highest output first",
        },
        Object {
          "func": [Function],
          "label": "Lowest output first",
        },
        Object {
          "func": [Function],
          "label": "Highest ECs first",
        },
        Object {
          "func": [Function],
          "label": "Lowest ECs first",
        },
        Object {
          "func": [Function],
          "label": "Highest fee first",
        },
        Object {
          "func": [Function],
          "label": "Lowest fee first",
        },
      ]
    }
  >
    <Component />
  </Sortable>
</div>
`;
